contents/tutorials/redirect-testing.md

```diff
@@ -0,0 +1,383 @@
+  return (
+    <html lang="en">
+      <PHProvider>
+        <body>{children}</body>
+      </PHProvider>
+    </html>
+  )
+}
+```
+
+Once set up, PostHog autocaptures usage and you can use all its tools throughout your app.
+
+## Adding test pages
+
+In the `app` folder, create two new folders named `control` and `test`. In each of them, create a basic `page.js` file with a button to capture an event. This is what the control page looks like:
+
+```js
+// app/control/page.js
+'use client'
+import { usePostHog } from "posthog-js/react";
+
+export default function Control() {
+  const posthog = usePostHog();
+
+  return (
+    <main>
+      <h1>Hello!</h1>
+      <button onClick={() => posthog.capture("main_button_clicked")}>
+        Click me!
+      </button>
+    </main>
+  );
+}
+```
+
+This is what the test page looks like:
+
+```js
+// app/test/page.js
+'use client'
+import { usePostHog } from "posthog-js/react";
+
+export default function Test() {
+  const posthog = usePostHog();
+
+  return (
+    <main>
+      <h1>Hello from the bright side!</h1>
+			<p>Clicking this button will bring you happiness</p>
+      <button onClick={() => posthog.capture("main_button_clicked")}>
+        I want to be happy!
+      </button>
+    </main>
+  );
+}
+```
+
+Now we can run `npm run dev`, go to each of our pages, `http://localhost:3000/control` and `http://localhost:3000/test`, to see that they work, and click the button to capture a custom event in PostHog (we need it to create our experiment).
+
+![Events](../images/tutorials/redirect-testing/events.png)
+
+## Creating our experiment
+
+Our A/B test compares these two pages to see which drives more button clicks. To do this, we go to the [experiment tab](https://app.posthog.com/experiments) in PostHog and click "New experiment." Choose a name, feature flag key (like `main-redirect`), set your experiment goal to `main_button_clicked`, and click "Save as draft."
+
+![A/B test set up](../images/tutorials/redirect-testing/test.png)
+
+Because we are working locally, you can click "Launch" right after and head back to your app to implement it. 
+
+## Setting up the redirect test middleware
+
+Next.js enables you to run middleware that intercepts and modifies requests for your app. We can use it to run our redirect test.
+
+To start, create a `middleware.js` file in the base `redirect-test` directory. Match both the `/test` and `/control` paths and have the `/test` path redirect to control as a placeholder.
+
+```js
+// redirect-test/middleware.js
+import { NextResponse } from 'next/server'
+
+export async function middleware(request) {
+  if (request.nextUrl.pathname=== '/test') {
+    return NextResponse.redirect(new URL('/control', request.url))
+  }
+  return NextResponse.next()
+}
+
+export const config = {
+  matcher: ['/test', '/control'],
+};
+```
+
+Next, we set up the logic for user IDs so we can target our test. We want the user experience to be consistent so we:
```

Comment:
do we need this? Can we target our test a different way? Like anonymous ids? 
I think this step is a bit of distraction from the main tutorial, and makes it harder to follow (since we're introducing a new concept)

Reply:
Oh I see now, it's because we need to make an API call to `decide` in the middleware. Still, this user ID bit threw me off. Can mention it in the beginning as part of what we going to do? Or perhaps there's a better way to make it easier to follow?

Reply:
Good point, rewritten.